core/src/cluster_info_vote_listener.rs

```diff
@@ -561,12 +598,23 @@ impl ClusterInfoVoteListener {
                     slot,
                 );
                 w_slot_tracker.total_stake += current_stake;
+                w_slot_tracker.gossip_only_stake += gossip_only_stake
             } else {
                 let mut total_stake = 0;
+                let mut gossip_only_stake = 0;
```

Comment:
@CriesofCarrots @jstarry I think the current gossip-only vote updates are actually pretty useful for debugging gossip latencies. With that in mind, I've introduced this `gossip_only_stake` tracker to support the old notification path that does not include votes from replay.

However I think this means I will need to plumb an alternate subscription model in RPC, maybe rename the current `NotificationEntry::Gossip -> NotificationEntry::GossipAndReplay` and then move this logic to `NotificationEntry::Gossip` only? Will I need to reduplicate all the subscription logic in `rpc_subscriptions.rs` as well?

Reply:
Yes, a `NotificationEntry` rename and a new `NotificationEntry` makes sense. But I think that we could reuse the `gossip_subscriptions` maps for both types of notifications as they are if we add a new commitment level to differentiate SingleGossip and SingleGossipAndReplay.
This can come in a follow-up pr, right? I would like to play around with that a little.